crates/typst-cli/src/package.rs

```diff
@@ -56,10 +59,84 @@ impl PackageStorage {
             if dir.exists() {
                 return Ok(dir);
             }
+            let parent_dir = dir.parent().unwrap();
+
+            let file_locking_err = |string| {
+                move |err| PackageError::FileLocking(eco_format!("{string}: {err}"))
+            };
```

Comment:
This should probably be a private top-level function.

Reply:
Why? It's only used in this one function, and it's very small for an anonymous function. So I think it should stay like that. At least for now. Maybe because of rebase or other changes, it will be used in multiple functions. Then it makes sense to put it in top-level scope.

Reply:
It's more about having it be a cheap top-level function instead of unnecessarily creating a closure instance on each call. It's possible this is done for us automatically on more optimized profiles, but it's preferable to avoid closures when you're not capturing anything in general.

This pattern is used elsewhere in the codebase to provide errors, containing just a single call to `eco_format!(...)`.

Reply:
I was just about to interject because of most likely inlined optimization of the compiler, but I don't know for sure if it does this.